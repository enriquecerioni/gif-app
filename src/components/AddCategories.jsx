import PropTypes from 'prop-types';
import { useState } from "react";

export const AddCategories = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");
  
  const onInputChange = (e) => {
    setInputValue( e.target.value );
  }
  
  const onSubmit = (e) => {
    e.preventDefault();
    if( inputValue.trim().length <= 1 ) return;
    
    onNewCategory(inputValue.trim());
    setInputValue('');
  }

  return (
        <form onSubmit={ onSubmit } aria-label="form" >
            <input 
            type="text" 
            placeholder="Buscar gifs..." 
            value={ inputValue }
            onChange={ onInputChange }
            />
        </form> 
    )
};

AddCategories.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}