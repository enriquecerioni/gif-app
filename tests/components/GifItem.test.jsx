import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Test <GifItem /> Component", () => {
  const title = "Goku";
  const url = "https://media.revistagq.com/photos/5f45010acb266484bb785c78/master/pass/dragon-ball-z.jpg";
  test("Debe coincidir con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar la imagen con el URL y el ALT indicado", () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    // expect( screen.getByRole('img').src ).toBe( url );
    // expect( screen.getByRole('img').title ).toBe( title );
    const { src, alt } = screen.getByRole('img');
    expect( src ).toBe( url );
    expect( alt ).toBe( title );  
  });

  test('Debe de mostrar el tÃ­tulo en el component', () => { 
    
    render(<GifItem title={title} url={url} />);
    expect( screen.getByText( title ) ).toBeTruthy();
    
   })
});
