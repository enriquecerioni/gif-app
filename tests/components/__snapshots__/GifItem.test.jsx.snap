// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifItem /> Component Debe coincidir con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://media.revistagq.com/photos/5f45010acb266484bb785c78/master/pass/dragon-ball-z.jpg"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
